xbmc/cores/DllLoader/ldt_keeper.c

```diff
@@ -49,7 +49,7 @@
 #ifdef  __cplusplus
 extern "C" {
 #endif
-#if defined(TARGET_ANDROID) && (defined(__i386__) || defined(__x86_64__)) && !defined(modify_ldt)
+#if defined(__linux__) && !defined(__GLIBC__) && !defined(modify_ldt)
```

Comment:
This changes semantics. Not good. Please emphasize.

Reply:
https://github.com/xbmc/xbmc/commit/a671cfd6d3db3cde3e4f99baa1f773e95975323d added the "!defined(modify_ldt)" codepath only for Android (and https://github.com/xbmc/xbmc/commit/fda4184fccfa2df1d46924c250407b6e03e8ab65 extended Android detection to x86_64). Kodi builds using musl need to use the same codepath because musl does not provide modify_ldt.
What about removing the limitation to Android and only check for "defined(modify_ldt)" instead? The code would look like this instead
```
#if defined(modify_ldt)
int modify_ldt(int func, void *ptr, unsigned long bytecount);
#else
#define modify_ldt(a,b,c) syscall( __NR_modify_ldt,  a, b, c);
#endif
```
and builds with musl. I will check with uClibc & glibc before updating this PR.
A define to specifically check for musl does not exist: https://www.openwall.com/lists/musl/2013/03/29/13